src/System.Device.Gpio/System/Device/Gpio/GpioController.cs

```diff
@@ -59,25 +69,26 @@ public GpioController(PinNumberingScheme numberingScheme, GpioDriver driver)
         /// <summary>
         /// The number of pins provided by the controller.
         /// </summary>
-        public int PinCount => _driver.PinCount;
+        public virtual int PinCount => _driver.PinCount;
 
         /// <summary>
         /// Gets the logical pin number in the controller's numbering scheme.
         /// </summary>
-        /// <param name="pinNumber">The pin number in the controller's numbering scheme.</param>
+        /// <param name="pinNumber">The pin number</param>
+        /// <param name="givenScheme">The scheme for the pin number</param>
         /// <returns>The logical pin number in the controller's numbering scheme.</returns>
-        private int GetLogicalPinNumber(int pinNumber)
+        protected virtual int GetLogicalPinNumber(int pinNumber, PinNumberingScheme givenScheme)
```

Comment:
why cannot this use NumberingScheme property instead of argument?

Reply:
Because it is sometimes required to do a transform from a scheme that is not necessarily the default, see the `Dispose` method. It would also be possible (and has been suggested already) to add overloads to `Open(), Write() or Read()` that take the numbering scheme as extra argument, so that the default can be overriden for individual calls.

Reply:
Discussed directly and proposed alternate approach